spec-tests/src/main/java/org/candlepin/spec/bootstrap/client/StandardExportFactory.java

```diff
@@ -0,0 +1,193 @@
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.spec.bootstrap.client;
+
+import org.candlepin.ApiException;
+import org.candlepin.dto.api.v1.BrandingDTO;
+import org.candlepin.dto.api.v1.ConsumerDTO;
+import org.candlepin.dto.api.v1.ContentDTO;
+import org.candlepin.dto.api.v1.PoolDTO;
+import org.candlepin.dto.api.v1.ProductDTO;
+import org.candlepin.dto.api.v1.ReleaseVerDTO;
+import org.candlepin.resource.CdnApi;
+import org.candlepin.resource.OwnerContentApi;
+import org.candlepin.resource.OwnerProductApi;
+import org.candlepin.resource.RolesApi;
+import org.candlepin.resource.UsersApi;
+import org.candlepin.spec.bootstrap.client.api.ConsumerClient;
+import org.candlepin.spec.bootstrap.client.api.JobsClient;
+import org.candlepin.spec.bootstrap.client.api.OwnerClient;
+import org.candlepin.spec.bootstrap.data.builder.Cdns;
+import org.candlepin.spec.bootstrap.data.builder.Contents;
+import org.candlepin.spec.bootstrap.data.builder.Pools;
+import org.candlepin.spec.bootstrap.data.builder.ProductAttributes;
+import org.candlepin.spec.bootstrap.data.builder.Products;
+import org.candlepin.spec.bootstrap.data.util.StringUtil;
+
+import java.time.OffsetDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A factory for building standard manifest exports. Creates all of the needed products, content,
+ * and pools for generating a standard manifest.
+ */
+public class StandardExportFactory extends AbstractExportFactory {
+    private OwnerProductApi ownerProductApi;
+    private OwnerContentApi ownerContentApi;
+    private CdnApi cdnApi;
+
+    public StandardExportFactory(ConsumerClient consumerClient, OwnerClient ownerClient,
+        OwnerProductApi ownerProductApi, UsersApi userApi, RolesApi rolesApi,
+        OwnerContentApi ownerContentApi, CdnApi cdnApi, JobsClient jobsClient) throws ApiException {
+        super(consumerClient, ownerClient, userApi, rolesApi, jobsClient);
+        if (ownerProductApi == null) {
+            throw new IllegalArgumentException("OwnerProductApi must not be null.");
+        }
+        if (ownerContentApi == null) {
+            throw new IllegalArgumentException("OwnerContentApi must not be null.");
+        }
+        if (cdnApi == null) {
+            throw new IllegalArgumentException("CdnApi must not be null.");
+        }
+
+        this.ownerProductApi = ownerProductApi;
+        this.ownerContentApi = ownerContentApi;
+        this.cdnApi = cdnApi;
+
+        initialize();
+    }
+
+    private void initialize() throws ApiException {
```

Comment:
I created methods bindPoolsToConsumer, createPoolsForProducts, and createPool in the AbstractExportFactory class to help with the repetitive code.

Reply:
My apologies here, I was meaning to abstract some of the repeated blocks of code into a separate private method in this class, not making some of the behavior more generic and exposed externally.

For instance, some of it could be abstracted into like a "createVDCProduct" method which creates the content, SKU and engineering entities. However, since it's all one-off interconnected stuff, so sometimes that's hard to break up; and if not broken up properly, it can get even worse to read than a long method. I'll leave the judgement up to you, but the protected methods in the parent class isn't what I had in mind.

Reply:
That makes sense! Like you said some of test data initialization is hard to break up. The createVDCProduct makes sense and I added it. I moved the bindPoolsToConsumer, createPoolsForProducts, and createPool methods to private methods in the ExportSpecTest.